src/ast/passes/semantic_analyser.cpp

```diff
@@ -2300,7 +2276,7 @@ void SemanticAnalyser::visit(ArrayAccess &arr)
   const SizedType &type = arr.expr.type();
 
   if (is_final_pass()) {
-    if (!type.IsArrayTy() && !type.IsPtrTy()) {
+    if (!type.IsArrayTy() && !type.IsPtrTy() && !type.IsStringTy()) {
```

Comment:
This is IMHO a considerable change. We should consider:
- adding a CHANGELOG entry
- adding tests
- updating the below error message.

Perhaps it would deserve to be in a separate commit, it will be easier to locate if it breaks anything.

Reply:
I will split this out and do the above.